### PR TITLE
Fix: set Cayley as default if orthogonal_map is None

### PR DESCRIPTION
**Pull request by lucas-heck**
Current code passes `orthogonal_map=None` to `torch.nn.utils.parametrizations.orthogonal` in the BiMap. This PR changes this by instead having `orthogonal_map="cayley"` if BiMap uses None.

When using a squared BiMap the current code work fine because pytorch will default to `orthogonal_map=matrix_exp`, but if using a rectangular one then it uses `orthogonal_map=householder`. This has some issues when using weight decay which can cause `Q=0`, consequently `W=0`,and BiMap then outputs a zero matrix.

The current change can be seen when running the `moabb_hydra_benchmark.py` before and after the change:

| Model     | Accuracy (None) -> (cayley) |
| --------- | --------------------------: |
| SPDNet    |            0.6493 -> 0.6493 |
| TSMNet    |            0.2500 -> 0.6562 |
| EEGSPDNet |            0.2500 -> 0.5729 |

This also led to a nice speedup in my machine (ran using pytest-benchmark):

| Name (time in s)                         | cayley | None |  ΔMean |
| ---------------------------------------- | ---------------: | ----------------: | -----: |
| test_correctness_spd_learn[PhaseSPDNet]  |           1.8265 |            1.8220 |  -0.2% |
| test_correctness_spd_learn[TSMNet]       |           2.7115 |            2.7772 |  +2.4% |
| test_correctness_spd_learn[TensorCSPNet] |          11.5278 |           12.9794 | +12.6% |
| test_correctness_spd_learn[Green]        |          20.4135 |           21.3182 |  +4.4% |
| test_correctness_spd_learn[EEGSPDNet]    |          37.1012 |           49.6138 | +33.7% |

### CODE REVIEW

**Comment by bruAristimunha** [comment]
thanks @lucas-heck 🙏🏽
